tests/Unit/Projection/DefaultProjectionRepositoryTest.php

```diff
@@ -128,6 +131,77 @@ public function drop(): void
         $projectionRepository->handle($profileCreated);
     }
 
+    public function testHandleWithAttributes(): void
+    {
+        $projection = new class implements Projection {
+            use AttributeHandleMethod;
+
+            public static ?AggregateChanged $handledEvent = null;
+
+            #[Handle(ProfileCreated::class)]
+            public function handleProfileCreated(ProfileCreated $event): void
+            {
+                self::$handledEvent = $event;
+            }
+
+            public function create(): void
+            {
+            }
+
+            public function drop(): void
+            {
+            }
+        };
+
+        $profileCreated = ProfileCreated::raise(
+            ProfileId::fromString('1'),
+            Email::fromString('profile@test.com')
+        );
+
+        $projectionRepository = new DefaultProjectionRepository([$projection]);
+        $projectionRepository->handle($profileCreated);
+
+        self::assertSame($profileCreated, $projection::$handledEvent);
+
+        $profileVisited = ProfileVisited::raise(ProfileId::fromString('1'), ProfileId::fromString('2'));
+        $projectionRepository->handle($profileVisited);
```

Comment:
The assert if the method got handled is missing

Reply:
i rewrite the tests :)